appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/ApplicationProcessor.java

```diff
@@ -760,12 +756,63 @@ public void visitParameter(Parameter parameter, AnnotatedElement element, ApiCon
                         mediaType.setSchema(new SchemaImpl());
                     }
                     mediaType.getSchema()
-                            .setType(ModelUtils.mergeProperty(mediaType.getSchema().getType(), type, false));
+                    .setType(ModelUtils.mergeProperty(mediaType.getSchema().getType(), type, false));
                 }
             }
         }
     }
 
+    private static org.eclipse.microprofile.openapi.models.parameters.Parameter findOperationParameterFor(Parameter parameter,
+            Method annotated, ApiContext context) {
+        String name = parameter.name();
+        // If the parameter reference is valid
+        if (name != null && !name.isEmpty()) {
+            // Get all parameters with the same name
+            List<java.lang.reflect.Parameter> matchingMethodParameters = Arrays
+                    .asList(annotated.getParameters()).stream()
+                    .filter(x -> name.equals(ModelUtils.getParameterName(x)))
```

Comment:
FYI: swapped the equals because `ModelUtils.getParameterName(x)` can be null causeing NPE while `name` is from an annotation and cannot be null.